homeassistant/components/zwave_js/triggers/value_updated.py

```diff
@@ -0,0 +1,186 @@
+                {cc.value: cc.name for cc in CommandClass}
+            ),
+            vol.Required(ATTR_PROPERTY): vol.Any(vol.Coerce(int), cv.string),
+            vol.Optional(ATTR_ENDPOINT): vol.Coerce(int),
+            vol.Optional(ATTR_PROPERTY_KEY): vol.Any(vol.Coerce(int), cv.string),
+            vol.Optional(ATTR_FROM): vol.Any(VALUE_SCHEMA, [VALUE_SCHEMA]),
+            vol.Optional(ATTR_TO): vol.Any(VALUE_SCHEMA, [VALUE_SCHEMA]),
+        },
+    ),
+    cv.has_at_least_one_key(ATTR_ENTITY_ID, ATTR_DEVICE_ID),
+)
+
+
+async def async_attach_trigger(
+    hass: HomeAssistant,
+    config: ConfigType,
+    action: Callable,
+    automation_info: dict[str, Any],
+    *,
+    platform_type: str = PLATFORM_TYPE,
+) -> CALLBACK_TYPE:
+    """Listen for state changes based on configuration."""
+    nodes = set()
+    if ATTR_DEVICE_ID in config:
+        nodes.update(
+            {
+                async_get_node_from_device_id(hass, device_id)
+                for device_id in config.get(ATTR_DEVICE_ID, [])
+            }
+        )
+    if ATTR_ENTITY_ID in config:
+        nodes.update(
+            {
+                async_get_node_from_entity_id(hass, entity_id)
+                for entity_id in config.get(ATTR_ENTITY_ID, [])
+            }
+        )
+
+    from_value = config.get(ATTR_FROM, MATCH_ALL)
+    to_value = config.get(ATTR_TO, MATCH_ALL)
+    command_class = config[ATTR_COMMAND_CLASS]
+    property_ = config[ATTR_PROPERTY]
+    endpoint = config.get(ATTR_ENDPOINT)
+    property_key = config.get(ATTR_PROPERTY_KEY)
+    unsubs = []
+    job = HassJob(action)
+
+    trigger_data: dict = {}
+    if automation_info:
+        trigger_data = automation_info.get("trigger_data", {})
+
+    @callback
+    def async_on_value_updated(
+        value: Value, device: dr.DeviceEntry, event: Event
+    ) -> None:
+        """Handle value update."""
+        event_value: Value = event["value"]
+        if event_value != value:
+            return
+
+        # Get previous value and its state value if it exists
+        prev_value_raw = event["args"]["prevValue"]
+        prev_value = value.metadata.states.get(str(prev_value_raw), prev_value_raw)
+        # Get current value and its state value if it exists
+        curr_value_raw = event["args"]["newValue"]
+        curr_value = value.metadata.states.get(str(curr_value_raw), curr_value_raw)
+        # Check from and to values against previous and current values respectively
+        for value_to_eval, raw_value_to_eval, match in (
+            (prev_value, prev_value_raw, from_value),
+            (curr_value, curr_value_raw, to_value),
+        ):
+            if (
+                match != MATCH_ALL
+                and value_to_eval != match
+                and not (isinstance(match, list) and value_to_eval in match)
+                and raw_value_to_eval != match
+                and not (isinstance(match, list) and raw_value_to_eval in match)
```

Comment:
We already checked the match list condition above.

Reply:
I can combine these checks but the second part of the check is checking against different values, the raw and the translated value